api-cursors/src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -254,11 +264,16 @@ private void validateSubscriptionCommitCursors(final Subscription subscription,
     }
 
     private void validateCursorsBelongToSubscription(final Subscription subscription,
-                                                     final List<NakadiCursor> cursors)
+                                                     final List<NakadiCursor> cursors,
+                                                     final Collection<String> additionalAllowedEventTypes)
             throws UnableProcessException {
-        final List<String> wrongEventTypes = cursors.stream()
+        var stream = cursors.stream()
```

Comment:
It looks like having the event type checked all the time will reduce amount of code. I would just add it by default and check it for every subscription in `validateCursorsBelongToSubscription()` method. It will be smaller change like that.

```
  final List<String> wrongEventTypes = cursors.stream().
  .filter(et -> !subscription.getEventTypes().contains(et))
  .filter(et -> !et.equals(dlqEventType))
  .collect(Collectors.toList());
```

The actual correct way to do it would be to check the topology from the zookeeper. That can be done by refactoring validateStreamId: separating the logic of `getTopology()` into different method. Once you have a topology you can execute necessary check like if dlqEventType in topology then enrich Subscription with dlqEventType by creating ephemeral subscription and reusing the logic `validateCursorsBelongToSubscription`, it will be easier to understand / read after a couple of months.

Reply:
+1 for checking cursors against the subscription's topology (vs. list of event types in subscription).

Reply:
are you going to add that ?

Reply:
This is already added in the latest PR update.